server/routes/induction/update/previousWorkExperienceDetailUpdateController.ts

```diff
@@ -0,0 +1,115 @@
+import createError from 'http-errors'
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { InductionDto, PreviousWorkExperienceDto } from 'inductionDto'
+import type { PreviousWorkExperienceDetailForm } from 'inductionForms'
+import { InductionService } from '../../../services'
+import PreviousWorkExperienceDetailController from '../common/previousWorkExperienceDetailController'
+import toCreateOrUpdateInductionDto from '../../../data/mappers/createOrUpdateInductionDtoMapper'
+import logger from '../../../../logger'
+import validatePreviousWorkExperienceDetailForm from './previousWorkExperienceDetailFormValidator'
+import TypeOfWorkExperienceValue from '../../../enums/typeOfWorkExperienceValue'
+
+/**
+ * Controller for the Update of the Previous Work Experience Detail screen of the Induction.
+ */
+
+export default class PreviousWorkExperienceDetailUpdateController extends PreviousWorkExperienceDetailController {
+  constructor(private readonly inductionService: InductionService) {
+    super()
+  }
+
+  getBackLinkUrl(req: Request): string {
+    const { prisonNumber } = req.params
+    return `/plan/${prisonNumber}/view/work-and-interests`
+  }
+
+  getBackLinkAriaText(_req: Request): string {
+    return 'Back to <TODO - check what CIAG UI does here bearing in mind the previous page might be the create journey or the update journey for either a specific job type, or all the job types>'
+  }
+
+  submitPreviousWorkExperienceDetailForm: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ): Promise<void> => {
+    const { prisonNumber } = req.params
+    const { prisonerSummary, inductionDto } = req.session
+    const { prisonId } = prisonerSummary
+    const { typeOfWorkExperience } = req.params
+
+    const previousWorkExperienceType = Object.values(TypeOfWorkExperienceValue).find(
+      type => type === typeOfWorkExperience.toUpperCase(),
+    )
+    if (!previousWorkExperienceType) {
+      logger.warn(
+        `Attempt to submit PreviousWorkExperienceDetailForm for invalid work experience type ${typeOfWorkExperience} from ${prisonerSummary.prisonNumber}'s Induction`,
+      )
+      return next(createError(404, `Previous Work Experience type ${typeOfWorkExperience} not found on Induction`))
+    }
+
+    const previousWorkExperiencesOnInduction = (inductionDto as InductionDto).previousWorkExperiences.experiences.map(
+      experience => experience.experienceType,
+    )
+    if (!previousWorkExperiencesOnInduction.includes(previousWorkExperienceType)) {
```

Comment:
Minor - is it worth combining this if statement with the one above? It's just that there are quite a few lines devoted to validation, so it might be good to reduce it.